src/libs/DateUtils.js

```diff
@@ -33,32 +34,48 @@ function getLocalMomentFromTimestamp(timestamp) {
  * Jan 20 at 5:30 PM          within the past year
  * Jan 20, 2019 at 5:30 PM    anything over 1 year ago
  *
+ * @param {String} locale
  * @param {Number} timestamp
  * @param {Boolean} includeTimeZone
  *
  * @returns {String}
  */
-function timestampToDateTime(timestamp, includeTimeZone = false) {
-    const date = getLocalMomentFromTimestamp(timestamp);
-    const isThisYear = moment().format('YYYY') === date.format('YYYY');
-    const isToday = moment().format('D MMM YYYY') === date.format('D MMM YYYY');
-    const yesterday = moment().add(-1, 'day').format('D MMM YYYY');
-    const isYesterday = yesterday === date.format('D MMM YYYY');
+function timestampToDateTime(locale, timestamp, includeTimeZone = false) {
+    const date = getLocalMomentFromTimestamp(locale, timestamp);
 
-    let format = 'LT';
-    if (isYesterday) {
-        format = `[Yesterday at] ${format}`;
-    } else if (!isToday) {
-        format = `MMM D [at] ${format}`;
-    } else if (!isThisYear) {
-        format = `MMM D, YYYY [at] ${format}`;
-    }
+    moment.calendarFormat = function (myMoment, now) {
+        const diff = myMoment.diff(now, 'days', true);
 
-    if (includeTimeZone) {
-        format = `${format} [UTC]Z`;
-    }
+        let retVal = 'sameElse';
 
-    return date.format(format);
+        if (diff < -1) {
+            retVal = 'lastWeek';
+        } else if (diff < 0) {
+            retVal = 'lastDay';
+        } else if (diff < 1) {
+            retVal = 'sameDay';
+        } else if (diff < 2) {
+            retVal = 'nextDay';
+        } else if (diff < 7) {
+            retVal = 'nextWeek';
+        }
+
+        if (includeTimeZone) {
+            retVal = 'timeZone';
```

Comment:
Is this going to be equivalent to what we had? Before we appended the timezone to the format ie: `Yesterday at 10:20 PST` but now it seems we will return a full date when the `includeTimeZone` param is true, no?

Reply:
I _think_ that what we want is:
- Move this override outside this function
- Remove the `timeZone` handling from it
- After computing the calendar date, if `includeTimeZone` was passed, append the timezone to the generated string.

Right?

Reply:
Aaahh, you are correct! Thanks, will make the changes and push.

Reply:
Updated :)